go/runtime/txpool/txpool.go

```diff
@@ -779,24 +820,52 @@ func (t *txPool) recheckWorker() {
 		case <-t.recheckTxCh.Out():
 		}
 
-		// Get a batch of scheduled transactions.
-		txs := t.schedulerQueue.getAll()
+		t.recheck()
+	}
+}
 
-		if len(txs) == 0 {
-			continue
-		}
+func (t *txPool) recheck() {
+	t.drainLock.Lock()
```

Comment:
So rechecking now blocks scheduling? Could this be a problem? I guess only if the current node is a scheduler/proposer and the recheck is taking long (timeout is 15 sec).

Reply:
it does.

makes sense that it could be a problem if the queue gets so big. although the time to check the max queue size worth of transactions is less than 15 seconds right?

maybe if we need even longer queues, we could have some way to recheck them in batches and unblock the scheduling as soon as the highest priority ones are done